### DISCUSSION TITLE
How to add prefix to baseColor_xx.jpg?

### DISCUSSION THREAD

**Discussion by ringcrl** [donmccurdy/glTF-Transform#1353] [Q&A] [answered]
After using the write method, there will be three files exported: `baseColor.jpg`, `modelName.bin`, and `modelName.gltf`. However, all my resources are stored in one directory. Different models will use the same `baseColor.jpg` file, but they actually have different contents, which may cause them to overwrite each other. I haven't found a place where I can configure the name "baseColor". Please let me know how can I make the exported texture image in the form of `modelName.baseColor.jpg`, thank

**Comment by donmccurdy** [comment, marked as answer]
Hi @ringcrl! It sounds like you are using the scripting API and `NodeIO` to read/write? If that's not the case and you're using the CLI instead, I think you'll need to use the scripting API. The names of textures are configured by the [`Texture#setURI`](https://gltf-transform.dev/modules/core/classes/Texture#setURI) method; if unspecified, defaults are chosen.

For example:

```javascript
import { ImageUtils } from '@gltf-transform/core';
import { listTextureSlots } from '@gltf-transform/functions';

const io = new NodeIO();
const document = await io.read('path/to/modelName.glb');
const prefix = 'modelName';

for (const texture of document.getRoot().listTextures()) {
  const slot = listTextureSlots(texture)[0] || 'unused';
  const extension = ImageUtils.mimeTypeToExtension(texture.getMimeType());
  texture.setURI(`${myPrefix}.${slot}.${extension}`);
}

await io.write('path/to/modelName.gltf', document);
```

If you prefer to put images into folders, you can also prefix the path with the name of a folder.

**Reply by ringcrl** [reply]
It is work as expect！thanks a lot！
